test/run-smoke-tests.sh

```diff
@@ -44,7 +44,7 @@ while [ $# != 0 ] ; do
 	esac
 done
 
-rustup update
+rustup update --no-self-update "$(rustup show active-toolchain|cut -d' ' -f1)"
```

Comment:
Maybe consider leaving a comment here why this is important so that nobody (including me) tries to change this in the future?